core/dmux.go

```diff
@@ -311,7 +378,172 @@ func simpleSetup(size, qsize int, sink Sink) ([]chan interface{}, *sync.WaitGrou
 		go func(index int) {
 			sk := sink.Clone()
 			for msg := range ch[index] {
-				sk.Consume(msg)
+				sk.Consume(msg, math.MaxInt32)
+			}
+			wg.Done()
+		}(i)
+	}
+	return ch, wg
+}
+
+func simpleSetupWithSideline(size, qsize int, sink Sink, sideline Sideline, sidelineImpl interface{}) ([]chan interface{}, *sync.WaitGroup) {
+	wg := new(sync.WaitGroup)
+	wg.Add(size)
+	ch := make([]chan interface{}, size)
+	sidelineChannel := make([]chan ChannelObject, size)
+	sinkChannel := make([]chan ChannelObject, size)
+
+	for i := 0; i < size; i++ {
+		sinkChannel[i] = make(chan ChannelObject, qsize)
+		go func(index int) {
```

Comment:
Extract out to an external function, to avoid redefinition of anon function

Reply:
will change this